CMake/stb/main.cpp

```diff
@@ -0,0 +1,11 @@
+
```

Comment:
It could be worth adding some comments describing what we are trying to do here:

```suggestion
// On some older STB versions, `STB_IMAGE_STATIC` does not correctly mark all
// functions as static.  So here we include two header files that in turn
// include `STB_IMAGE_IMPLEMENTATION` and `STB_IMAGE_STATIC` to check.  If
// linking fails, our STB version is not new enough.
```